src/metabase/channel/email.clj

```diff
@@ -86,19 +88,40 @@
 ;; ------------------------------------------------------------------------------------------------;;
 
 (mu/defmethod channel/render-notification [:channel/email :notification/dashboard-subscription] :- [:sequential EmailMessage]
-  [_channel-type {:keys [dashboard payload pulse]} recipients]
+  [_channel-type {:keys [dashboard payload pulse]} _template recipients]
   (let [{:keys [user-emails
                 non-user-emails]} (recipients->emails recipients)
         timezone                  (some->> payload (some :card) channel.shared/defaulted-timezone)
         email-subject             (:name dashboard)
         email-to-users            (when (seq user-emails)
-                                    (construct-pulse-email
+                                    (construct-email
                                      email-subject
                                      user-emails
                                      (messages/render-pulse-email timezone pulse dashboard payload nil)))
         email-to-nonusers         (for [non-user-email non-user-emails]
-                                    (construct-pulse-email
+                                    (construct-email
                                      email-subject
                                      [non-user-email]
                                      (messages/render-pulse-email timezone pulse dashboard payload non-user-email)))]
     (filter some? (conj email-to-nonusers email-to-users))))
+
+;; ------------------------------------------------------------------------------------------------;;
+;;                                         System Events                                           ;;
+;; ------------------------------------------------------------------------------------------------;;
+
+(defn- notification-recipients->emails
+  [recipients]
+  (for [recipient recipients]
+    (case (:type recipient)
+      :notification-recipient/user
+      (-> recipient :user :email))))
```

Comment:
We want this to go *boom* for unknown types? Perhaps it should log a warning instead.

Reply:
yep, I plan to do that in followup PRs